tests/python/unittest/test_gluon_rnn.py

```diff
@@ -890,152 +657,53 @@ def __init__(self, rnn_size, time_step, **kwargs):
-                      mx.nd.ones((8, 3, 5, 7)))
+                      mx.np.ones((8, 3, 5, 7)))
     check_rnn_forward(gluon.rnn.Conv1DGRUCell((5, 7), 10, (3,), (3,)),
-                      mx.nd.ones((8, 3, 5, 7)))
+                      mx.np.ones((8, 3, 5, 7)))
 
     net = mx.gluon.rnn.SequentialRNNCell()
     net.add(gluon.rnn.Conv1DLSTMCell((5, 7), 10, (3,), (3,)))
     net.add(gluon.rnn.Conv1DRNNCell((10, 5), 11, (3,), (3,)))
     net.add(gluon.rnn.Conv1DGRUCell((11, 3), 12, (3,), (3,)))
-    check_rnn_forward(net, mx.nd.ones((8, 3, 5, 7)))
-
-
-def test_convrnn():
-    cell = gluon.rnn.Conv1DRNNCell((10, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 50), out_shape=(1, 100, 48))
-
-    cell = gluon.rnn.Conv2DRNNCell((10, 20, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 20, 50), out_shape=(1, 100, 18, 48))
-
-    cell = gluon.rnn.Conv3DRNNCell((10, 20, 30, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 20, 30, 50), out_shape=(1, 100, 18, 28, 48))
-
-
-def test_convlstm():
-    cell = gluon.rnn.Conv1DLSTMCell((10, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 50), out_shape=(1, 100, 48))
-
-    cell = gluon.rnn.Conv2DLSTMCell((10, 20, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 20, 50), out_shape=(1, 100, 18, 48))
-
-    cell = gluon.rnn.Conv3DLSTMCell((10, 20, 30, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 20, 30, 50), out_shape=(1, 100, 18, 28, 48))
-
-
-def test_convgru():
-    cell = gluon.rnn.Conv1DGRUCell((10, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 50), out_shape=(1, 100, 48))
-
-    cell = gluon.rnn.Conv2DGRUCell((10, 20, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 20, 50), out_shape=(1, 100, 18, 48))
-
-    cell = gluon.rnn.Conv3DGRUCell((10, 20, 30, 50), 100, 3, 3)
-    check_rnn_cell(cell, in_shape=(1, 10, 20, 30, 50), out_shape=(1, 100, 18, 28, 48))
-
-
-def test_conv_fill_shape():
-    cell = gluon.rnn.Conv1DLSTMCell((0, 7), 10, (3,), (3,))
-    cell.hybridize()
-    check_rnn_forward(cell, mx.nd.ones((8, 3, 5, 7)))
-    assert cell.i2h_weight.shape[1] == 5, cell.i2h_weight.shape[1]
-
-
-def test_lstmp():
-    nhid = 100
-    nproj = 64
-    cell = gluon.rnn.LSTMPCell(nhid, nproj)
-    inputs = [mx.sym.Variable('rnn_t%d_data'%i) for i in range(3)]
-    outputs, _ = cell.unroll(3, inputs)
-    outputs = mx.sym.Group(outputs)
-    expected_params = ['h2h_bias', 'h2h_weight', 'h2r_weight', 'i2h_bias', 'i2h_weight']
-    expected_outputs = [type(cell).__name__.lower() + name for name in ['_t0_out_output', '_t1_out_output', '_t2_out_output']]
-    assert sorted(cell.collect_params().keys()) == expected_params
-    assert outputs.list_outputs() == expected_outputs, outputs.list_outputs()
-
-    args, outs, auxs = outputs.infer_shape(rnn_t0_data=(10,50), rnn_t1_data=(10,50), rnn_t2_data=(10,50))
-    assert outs == [(10, nproj), (10, nproj), (10, nproj)]
-
-
-def test_vardrop():
```

Comment:
Should this test be adapted to testing the front-end infer_shape functions instead? Also, the correctness test should not be dropped.